src/System.Private.CoreLib/shared/System/Text/StringBuilderCache.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text
+{
+    /// <summary>Provide a cached reusable instance of stringbuilder per thread.</summary>
+    internal static class StringBuilderCache
+    {
+        // The value 360 was chosen in discussion with performance experts as a compromise between using
+        // as litle memory per thread as possible and still covering a large part of short-lived
+        // StringBuilder creations on the startup path of VS designers.
+        private const int MaxBuilderSize = 360;
```

Comment:
Note that corert and corefx both have a value of 260 rather than 360.  I've kept the 360 value.